test/onigumo_test.exs

```diff
@@ -70,6 +70,36 @@ defmodule OnigumoTest do
     assert(loaded_urls == @urls)
   end
 
+  test("get response by HTTP request") do
+    expect(HTTPoisonMock, :get!, &get!/1)
+
+    url = Enum.at(@urls, 0)
+    get_response = Onigumo.get_url(url, HTTPoisonMock)
+    expected_response = %HTTPoison.Response{
+      status_code: 200,
+      body: body(url)
+    }
```

Comment:
Can we leverage the `get!/1` function down there? The name is then not the best, but it composes the exact response we use here. Just as you do in the “extract body from URL response” test.